src/core/lib/security/credentials/tls/grpc_tls_credentials_options.cc

```diff
@@ -29,6 +29,44 @@
 
 #include "src/core/lib/surface/api_trace.h"
 
+int grpc_tls_credentials_options::cmp(
+    const grpc_tls_credentials_options* other_options) const {
+  GPR_ASSERT(other_options != nullptr);
+  if (cert_request_type_ != other_options->cert_request_type_) {
+    return 1;
```

Comment:
The `cmp()` function is not simply a boolean equality function; it's used for sorting.  It returns less than, equal to, or greater than 0 if the LHS is less than, equal to, or greater than the RHS.

So for each entry here, you need to do something like this:

```
int r = grpc_core::QsortCompare(cert_request_type_, other_options.cert_request_type_);
if (r != 0) return r;
```

Or, for string fields:

```
r = root_cert_name_.compare(other_options.root_cert_name_);
if (r != 0) return r;
```

Then, after comparing all fields, if you haven't yet returned, you can just return 0, because you know that all fields are equal.

Reply:
Sure. Just one quick question: is `grpc_core::QsortCompare` not capable of handling string type? I guess it can only handle primitive types like `enum`, `bool`, etc?

Reply:
Right, `grpc_core::QsortCompare()` only handles POD types, not strings.